Panels/PanelCanvasXYPlot.js

```diff
@@ -212,6 +212,23 @@ define([
                 return (vly - panel.offsetY ) / panel.scaleY;
             };
 
+            /**
+             * Tests whether part of the rectangular area around a point is visible in the viewport
+             * @param {float} vlx - the window x coordinate of the center point
+             * @param {float} vly - the window y coordinate of the center point
+             * @param {float} width - the window width of the rectangle centered at the point
+             * @param {float} height - the window height of the rectangle centered at the point
+             * @returns {bool} - whether part of the area is visible
+             */
+            panel.pointVisibleInViewport = function(vlx, vly, width, height) {
+                return (
+                    vlx - 0.5 * width >= 0 &&
+                    vlx + 0.5 * width <= panel._cnvWidth &&
+                    vly - 0.5 * height >= 0 &&
+                    vly + 0.5 * height <= panel._cnvHeight
+                );
```

Comment:
Only returns true if completely visible in viewPort, but I think we want to draw the point even if it only partially overlaps the viewport.

Reply:
Ah yes, of course!  Fixed the overlap test.